adapter/pkg/logging/log_format.go

```diff
@@ -0,0 +1,117 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package logging
+
+import (
+	"bytes"
+	"fmt"
+	"runtime"
+	"strings"
+
+	logrus "github.com/sirupsen/logrus"
+	"github.com/wso2/product-microgateway/adapter/pkg/config"
+)
+
+type plainFormatter struct {
+	TimestampFormat string
+	LevelDesc       []string
+}
+
+// ErrorHook for adding a custom logic to default the error details only for error logs
+type ErrorHook struct {
+}
+
+// LogFormatter is to keep the relevant log formatter
+var (
+	LogFormatter logrus.Formatter
+)
+
+func init() {
+	LoggerFormat()
+}
+
+// LoggerFormat will read the configs (LogFormalization) and decide which formatter to be used for logging
+func LoggerFormat() {
+	logConf := config.ReadLogConfigs()
+	switch logConf.LogFormalization {
+	case jsonType:
+		formatter := new(logrus.JSONFormatter)
+		formatter.TimestampFormat = "2006-01-02 15:04:05"
+		formatter.CallerPrettyfier = func(frame *runtime.Frame) (function string, file string) {
+			fileArr := strings.Split(frame.File, "/")
+			return formatFilePath(frame.Function), fileArr[len(fileArr)-1] + ":" + fmt.Sprintf("%d", frame.Line)
+		}
+		LogFormatter = formatter
+
+	default:
+		formatter := new(plainFormatter)
+		formatter.TimestampFormat = "2006-01-02 15:04:05"
+		formatter.LevelDesc = []string{
+			panicLevel,
+			fatalLevel,
+			errorLevel,
+			warnLevel,
+			infoLevel,
+			debugLevel}
+		LogFormatter = formatter
+	}
+}
+
+// Format sets a custom format for loggers.
+func (f *plainFormatter) Format(entry *logrus.Entry) ([]byte, error) {
+	timestamp := fmt.Sprintf(entry.Time.Format(f.TimestampFormat))
+	return []byte(fmt.Sprintf("%s %s [%s:%d] - [%s] [-] %s [%s]\n",
+		timestamp,
+		f.LevelDesc[entry.Level],
+		formatFilePath(entry.Caller.File),
+		entry.Caller.Line,
+		formatFilePath(entry.Caller.Function),
+		entry.Message,
+		createKeyValuePairs(entry.Data))), nil
+}
+
+func createKeyValuePairs(m logrus.Fields) string {
+	b := new(bytes.Buffer)
+	for key, value := range m {
+		fmt.Fprintf(b, "%s=%v ", key, value)
+	}
+	return strings.TrimSpace(b.String())
+}
+
+// formatFilePath retrieves only the last part from a path.
+func formatFilePath(path string) string {
+	arr := strings.Split(path, "/")
+	return arr[len(arr)-1]
+}
+
+// Levels to fire only on ErrorLevel (.Error(), .Errorf(), etc.)
+func (h *ErrorHook) Levels() []logrus.Level {
+	return []logrus.Level{logrus.ErrorLevel}
+}
+
+// Fire specifies a custom logic to execute when the hook fires
+func (h *ErrorHook) Fire(e *logrus.Entry) error {
+	// e.Data is a map with all fields attached to entry
+	if _, ok := e.Data["severity"]; !ok {
+		e.Data["severity"] = DEFAULT
+	}
+	if _, ok := e.Data["code"]; !ok {
+		e.Data["code"] = 0
```

Comment:
name "code" should be changed as "error_code"

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/2633/commits/c3087100180167139f73a7fb351a91359c5d9bbf